recipes/_start.rb

```diff
@@ -10,6 +10,16 @@
   not_if { platform_family?('windows') }
 end
 
+ruby_block 'insert ulimit' do
+  block do
+    file = Chef::Util::FileEdit.new('/etc/init.d/splunk')
+    file.insert_line_after_match(/^RETVAL=\d$/, "ulimit -n #{node['splunk']['limits']['open_files']}")
+    file.write_file
+  end
+  not_if { platform_family?('windows') || File.readlines('/etc/init.d/splunk').grep(/ulimit/).any? }
```

Comment:
Actually I could probably get rid of the greping for ulimit because, and Charlie can correct me if I am wrong, I believe the init.d script gets overwritten on each deploy so it would never have the ulimit added in the script.